http/js/script.js

```diff
@@ -79,7 +79,74 @@ function gaussian(x, a, b, c) {
     return a * Math.exp(- Math.pow(x - b, 2) / (2 * Math.pow(c, 2)));
 }
 
+/**
+ * Commit representation
+ */
+function Commit($tr) {
+    this.$tr = $tr;
+}
+Commit.prototype = {
+    $tr: null,
+
+    select: function() {
+        this.$tr.addClass('selected');
+    },
+    deselect: function() {
+        this.$tr.removeClass('selected');
+    }
+};
+
+/**
+ * Commits manager
+ */
+function CommitsManager(commitsTable) {
+    this.commitsTable = commitsTable;
+
+    // select the relevant commit if the URL says so
+    var hash = window.location.hash.substr(1);
+    if (hash.substr(0, 7) === 'commit-') {
+        var commit = this.find(hash.substr(7));
+        this.select(commit);
+    }
+
+    this.initEvents();
+}
+CommitsManager.prototype = {
+    commitsTable: null,
+    commits: [],
+    selectedCommit: null,
+
+    find: function(commitHash) {
+        if (!this.commits[commitHash]) {
+            var commitTr = this.commitsTable.find('tr#' + commitHash);
```

Comment:
Note: this could be a security issue if we ever handle sensitive data (or if another site on the same domain handles sensitive data!). I'm no expert on how to fix it though... needs some kind of validation.